server/routes/api.js

```diff
@@ -188,6 +228,32 @@ router.get('/templates/:id', (req, res) => {
   res.send(script);  
 });
 
+router.put('/templates/:id', (req, res) => {
+  let { id } = req.params;
+  let { script } = req.body || '';
```

Comment:
if req.body doesn't contain script. then putting '' is probably wrong.
why would you save an empty file?